### PR TITLE
Add testcases for rgw cloudtransition/cloudtiering feature

### PR DESCRIPTION
**Pull request by soumyakoduri**
Added few test cases for rgw cloud-transition/tiering feature. The sleep/waiting time in the tests may vary based on the I/O latency to the cloud endpoint.

Feature doc: https://docs.google.com/document/d/1IoeITPCF64A5W-UA-9Y3Vp2oSfz3xVQHu31GTu3u3Ug/edit#heading=h.4o1xyas8s23x

PR: ceph/ceph#35100

Also added existing boto2 lifecycle transition testcases to boto3

Signed-off-by: Soumya Koduri <skoduri@redhat.com>

### CODE REVIEW

**Comment by alimaredia** [comment]
Upon first glance the tests look good to me, but I'll leave @yehudasa (or someone else) to reviewing the specifics of test specific feedback.

**Comment by soumyakoduri** [comment]
> nice work on the test coverage. we'll want to get this running in teuthology with a separate rgw as the cloud endpoint
> 
> it would also be nice to test the behavior of these 'transitioned' objects with s3 operations like HEAD, GET, COPY, and DELETE

thanks.. sure. I will add these cases.

File: s3tests_boto3/functional/test_s3.py [hunk 1]
```diff
@@ -9828,6 +9855,511 @@ def _test_encryption_sse_customer_write(file_size):
...
+    time.sleep(100)
+    expire1_key1_str = prefix + keys[0]
+    verify_object(cloud_client, target_path, expire1_key1_str, keys[0], target_sc)
+
+    expire1_key2_str = prefix + keys[1]
+    verify_object(cloud_client, target_path, expire1_key2_str, keys[1], target_sc)
+
+    # Now verify the object on source rgw
+    src_key = keys[0]
+    if (retain_head_object != None and retain_head_object == "true"):
```
Comment (soumyakoduri) [inline s3tests_boto3/functional/test_s3.py:10124]: @cbodley have added cases to test Ops. Please review.

Comment (cbodley) [reply s3tests_boto3/functional/test_s3.py:10201]: looks great

**Review by cbodley — APPROVED** [review]

**Comment by alimaredia** [comment]
@soumyakoduri If you're porting tests from Boto2 -> Boto3 could you remove the boto2 version of the tests as well?

**Comment by soumyakoduri** [comment]
> @soumyakoduri If you're porting tests from Boto2 -> Boto3 could you remove the boto2 version of the tests as well?

Done.

**Comment by soumyakoduri** [comment]
> Upon first glance the tests look good to me, but I'll leave @yehudasa (or someone else) to reviewing the specifics of test specific feedback.
> 
> @soumyakoduri do you have a run in teuthology of these new tests?

Ran "rgw:multifs:s3tests" against this suite-branch -

http://pulpito.front.sepia.ceph.com/soumyakoduri-2021-12-06_09:31:42-rgw:multifs-lc_teuthology-distro-basic-smithi/

All the added testcases are skipped for now as expected.

**Comment by soumyakoduri** [comment]
These changes are merged as part of https://github.com/ceph/s3-tests/pull/421

**Pull request closed without merging**